Tests/scripts/hook_validations/docker.py

```diff
@@ -0,0 +1,239 @@
+from Tests.test_utils import get_yaml, print_error, server_version_compare
+from pkg_resources import parse_version
+from datetime import datetime, timedelta
+import re
+import requests
+
+
+# disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+ACCEPT_HEADER = {
+    'Accept': "application/json, "
+              "application/vnd.docker.distribution.manifest.v2+json, "
+              "application/vnd.docker.distribution.manifest.list.v2+json"
+}
+
+# use 10 seconds timeout for requests
+TIMEOUT = 10
+DEFAULT_REGISTRY = "registry-1.docker.io"
+
+
+def parse_www_auth(www_auth):
+    """Parse realm and service from www-authenticate string of the form:
+    Bearer realm="https://auth.docker.io/token",service="registry.docker.io"
+
+    :param www_auth: www-authenticate header value
+    :type www_auth: string
+    """
+    match = re.match(r'.*realm="(.+)",service="(.+)".*', www_auth, re.IGNORECASE)
+    if not match:
+        return None
+    return [match.group(1), match.group(2)]
+
+
+def docker_auth(image_name, verify_ssl=True, registry=DEFAULT_REGISTRY):
+    """
+    Authenticate to the docker service. Return an authentication token if authentication is required.
+    """
+    res = requests.get(
+        'https://{}/v2/'.format(registry),
+        headers=ACCEPT_HEADER,
+        timeout=TIMEOUT,
+        verify=verify_ssl
+    )
+    if res.status_code == 401:  # need to authenticate
+        # defaults in case we fail for some reason
+        realm = "https://auth.docker.io/token"
+        service = "registry.docker.io"
+        # Should contain header: Www-Authenticate
+        www_auth = res.headers.get('www-authenticate')
+        if www_auth:
+            parse_auth = parse_www_auth(www_auth)
+            if parse_auth and len(parse_auth) == 2:
+                realm, service = parse_auth
+        res = requests.get(
+            '{}?scope=repository:{}:pull&service={}'.format(realm, image_name, service),
+            headers=ACCEPT_HEADER,
+            timeout=TIMEOUT,
+            verify=verify_ssl
+        )
+        res.raise_for_status()
+        res_json = res.json()
```

Comment:
wrap in `try-except`

Reply:
already being wrapped in get_docker_image_latest_tag, this function handles all types of request exceptions.